Custom title and output path

mgerasimchuk opened the pull request:
…go-coverage-report/issues/10

mgerasimchuk committed:
Proposal solution for the feature request https://github.com/ncruces/go-coverage-report/issues/10

mgerasimchuk committed:
Add badge title customization param

mgerasimchuk commented:
Have tested on my own projects:
- jobs:
  - https://github.com/mgerasimchuk/protty/blob/master/.github/workflows/test-unit.yml#L11-L18
  - https://github.com/mgerasimchuk/protty/blob/master/.github/workflows/test-integration.yml#L11-L18
- wiki: https://github.com/mgerasimchuk/protty/wiki/Test-coverage-report
- readme: https://github.com/mgerasimchuk/protty/blob/master/README.md

*sorry for the unrelated changes in one MR, but I also added a possibility to change a badge title

ncruces commented on action.yml:
```diff
@@ -10,9 +10,14 @@ inputs:
     default: flat
+  badge-title:
+    description: Badge title generated by shields.io.
+    default: coverage
```
👍

ncruces commented on action.yml:
```diff
@@ -10,9 +10,14 @@ inputs:
     description: Generate an HTML coverage report.
     default: true
+  report-path:
+    description: Custom path for an HTML coverage report.
```
This isn't really a the report path.
It's actually the output folder where _all_ output files are created, right?

mgerasimchuk replied:
Yes, actually it is, I just didn't want to add something new, but if you have a suggestions, I'll fix it..

Anyway, I can rename param and ENVs like:
```
output-path:
    description: Custom path for an HTML coverage report and other results file like svg badge and chart.
```

What do you think?

mgerasimchuk replied:
fixed

ncruces reviewed (commented):
Thanks for the PR!
Tbh I'm not 100% convinced about this…

mgerasimchuk committed:
Rename report-path to output-path

mgerasimchuk committed:
Fix concurrency pushes conflict by adding attempt and pull with rebase

If the several action working at the same time, and try to push the wiki changes it leads to the git remote rejected error. This commit adds the pull operation if there are remote rejected errors exists

ncruces commented on action.yml:
```diff
@@ -10,9 +10,14 @@ inputs:
   report:
     description: Generate an HTML coverage report.
     default: true
+  output-path:
```
I'll probably change this to `output-dir`, for extra clarity.

ncruces commented on action.yml:
```diff
@@ -61,5 +67,5 @@ runs:
         test ${{inputs.amend}} == true   && \
-          git commit --amend --no-edit   && git push --force-with-lease || \
-          git commit -m"Update coverage" && git push
+          git commit --amend --no-edit   && for i in {1..10}; do git push --force-with-lease && c=0 && break || c=$? && git pull --rebase; done; (exit $c) || \
```
Love the retry logic… but it's getting to the point that I need to create a proper shell script, I guess.

Will do it as a follow up.
